.circleci/config.yml

```
@@ -1,56 +1,129 @@
+          command: |
+            # Remove expired DST Root CA X3 certificate. Workaround
+            # for https://debbugs.gnu.org/cgi/bugreport.cgi?bug=51038
+            # bug on Emacs 27.2.
+            gci cert:\LocalMachine\Root\DAC9024F54D8F6DF94935FB1732638CA6AD77C13
+            gci cert:\LocalMachine\Root\DAC9024F54D8F6DF94935FB1732638CA6AD77C13 | Remove-Item
+            (iwr https://raw.github.com/doublep/eldev/master/webinstall/circle-eldev.ps1).Content | powershell -command -
+  test:
+    steps:
+      - run:
+          name: Run regression tests
+          command: eldev -dtT -p test
+  lint:
+    steps:
+      - run:
+          name: Lint
+          command: eldev lint -c
+  compile:
+    steps:
+      - run:
+          name: Check for byte-compilation errors
+          command: eldev -dtT compile --warnings-as-errors
 
-# Enumerated list of Emacs versions
 jobs:
-  test-emacs-25:
+  test-ubuntu-emacs-26:
     docker:
-      - image: silex/emacs:25-ci-cask
+      - image: silex/emacs:26-ci
         entrypoint: bash
-    <<: *default-steps
-
-  test-emacs-26:
+    steps:
+      - setup
+      - test
+      - lint
+      - compile
+  test-ubuntu-emacs-27:
     docker:
-      - image: silex/emacs:26-ci-cask
+      - image: silex/emacs:27-ci
         entrypoint: bash
-    <<: *default-steps
-
-  test-emacs-27:
+    steps:
+      - setup
+      - test
+      - lint
+      - compile
+  test-ubuntu-emacs-28:
     docker:
-      - image: silex/emacs:27-ci-cask
+      - image: silex/emacs:28-ci
         entrypoint: bash
-    <<: *default-steps
-
-  test-emacs-28:
+    steps:
+      - setup
+      - test
+      - lint
+      - compile
+  test-ubuntu-emacs-29:
     docker:
-      - image: silex/emacs:28-ci-cask
+      - image: silex/emacs:29-ci
         entrypoint: bash
-    <<: *default-steps
-
-  test-emacs-master:
+    steps:
+      - setup
+      - test
+      - lint
+      - compile
+  test-ubuntu-emacs-master:
     docker:
-      - image: silex/emacs:master-ci-cask
+      - image: silex/emacs:master-ci
         entrypoint: bash
-    <<: *default-steps
+    steps:
+      - setup
+      - test
+      - lint
+      - compile
+  test-macos-emacs-latest:
+    macos:
+      xcode: "14.2.0"
+    steps:
+      - setup-macos
+      - test
+      - lint
+      - compile
+  test-windows-emacs-latest:
+    executor: win/default
+    steps:
+      - run:
+          name: Install Emacs latest
+          command: |
+            choco install emacs
+      - setup-windows
+      - test
+      - lint
+      - compile
 
 workflows:
-  version: 2
+  version: 2.1
   ci-test-matrix:
     jobs:
-      - test-emacs-25
-      - test-emacs-26
-      - test-emacs-27
-      - test-emacs-28
-      - test-emacs-master
+      - test-ubuntu-emacs-26
+      - test-ubuntu-emacs-27
+      - test-ubuntu-emacs-28
+      - test-ubuntu-emacs-29
+      - test-ubuntu-emacs-master
+      - test-windows-emacs-latest
+      - test-macos-emacs-latest
```

```suggestion
      - test-macos-emacs-latest:
          requires:
            - test-ubuntu-emacs-29
```

macOS builds are little more precious